core/src/plugins/filed/python/vmware/BareosFdPluginVMware.py

```diff
@@ -3375,6 +3384,70 @@ def _transform_virtual_floppy(self, device):
 
         return add_device
 
+    def _transform_virtual_pci_passthrough(self, device):
+        add_device = vim.vm.device.VirtualPCIPassthrough()
+        add_device.key = device["key"] * -1
+        if (
+            device["backing"]["_vimtype"]
+            == "vim.vm.device.VirtualPCIPassthrough.DeviceBackingInfo"
+        ):
+            add_device.backing = vim.vm.device.VirtualPCIPassthrough.DeviceBackingInfo()
+            add_device.backing.deviceId = device["backing"]["deviceId"]
+            add_device.backing.deviceName = device["backing"]["deviceName"]
+            add_device.backing.id = device["backing"]["id"]
+            add_device.backing.systemId = device["backing"]["systemId"]
+            add_device.backing.useAutoDetect = device["backing"]["useAutoDetect"]
+            add_device.backing.vendorId = device["backing"]["vendorId"]
+        else:
+            raise RuntimeError(
+                "Unknown Backing for VirtualPCIPassthrough: %s"
+                % (device["backing"]["_vimtype"])
+            )
+            return None
+
+        if device["connectable"]:
+            add_device.connectable = self._transform_connectable(device)
+
+        # Looks like negative values must not be used for default devices,
+        # the VirtualPCIController always seems to have key 100.
+        add_device.controllerKey = device["controllerKey"]
+        if device["controllerKey"] != 100:
+            add_device.controllerKey = device["controllerKey"] * -1
+
+        add_device.unitNumber = device["unitNumber"]
```

Comment:
This block is the same as line 3438 and following. Maybe that can be somehow unified?

Reply:
Good idea, it's unified now.